eitprocessing/roi_selection/gridselection.py

```diff
@@ -0,0 +1,390 @@
+            # nan elements between numeric elements are kept
+            numeric_element_indices = np.argwhere(~np.isnan(vector))
+            first_num_element = numeric_element_indices.min()
+            last_num_element = numeric_element_indices.max()
+        else:
+            first_num_element = 0
+            last_num_element = len(vector) - 1
+
+        n_elements = last_num_element - first_num_element + 1
+
+        group_size = n_elements / n_groups
+
+        if group_size < 1:
+            if horizontal:
+                warnings.warn(
+                    f"The number horizontal regions ({n_groups}) is larger than the "
+                    f"number of available columns ({n_elements}).",
+                    MoreHorizontalGroupsThanColumns,
+                )
+            else:
+                warnings.warn(
+                    f"The number vertical regions ({n_groups}) is larger than the "
+                    f"number of available rows ({n_elements}).",
+                    MoreVerticalGroupsThanRows,
+                )
+
+        # find the right boundaries (upper values) of each group
+        right_boundaries = (np.arange(n_groups) + 1) * group_size
+        right_boundaries = right_boundaries[:, np.newaxis]  # converts to row vector
+
+        # each row in the base represents one group
+        base = np.tile(np.arange(n_elements), (n_groups, 1))
+
+        # if the element number is higher than the split, it does not belong in this group
+        element_contribution_to_group = right_boundaries - base
+        element_contribution_to_group[element_contribution_to_group < 0] = 0
+
+        # if the element to the right is a full group size, this element is ruled out
+        rule_out = element_contribution_to_group[:, 1:] >= group_size
+        element_contribution_to_group[:, :-1][rule_out] = 0
+
+        # elements have a maximum value of 1
+        element_contribution_to_group = np.fmin(element_contribution_to_group, 1)
+
+        # if this element is already represented in the previous group (row), subtract that
+        element_contribution_to_group[1:] -= element_contribution_to_group[:-1]
+        element_contribution_to_group[element_contribution_to_group < 0] = 0
+
+        # element_contribution_to_group only represents non-nan elements
+        # insert into final including non-nan elements
+        final = np.full((n_groups, len(vector)), np.nan)
+        final[
+            :, first_num_element : last_num_element + 1
+        ] = element_contribution_to_group
+
+        # convert to list of vectors
+        final = [final[n, :] for n in range(final.shape[0])]
+
+        return final
+
+    def matrix_layout(self) -> NDArray:
+        """Returns a 2D array showing the layout of the matrices returned by
+        `find_grid`."""
+        n_regions = self.v_split * self.h_split
+        return np.reshape(np.arange(n_regions), (self.v_split, self.h_split))
```

Comment:
I think it'd be nice to make this a property rather than a function. See 33905ee for suggestion